nixos/modules/system/boot/stage-1-init.sh

```diff
@@ -221,6 +221,9 @@ checkFS() {
     # Don't check resilient COWs as they validate the fs structures at mount time
     if [ "$fsType" = btrfs -o "$fsType" = zfs ]; then return 0; fi
 
+    # Skip fsck for bcachefs - not implemented yet.
+    if [ "$fsType" = bcachefs ]; then return 0; fi
```

Comment:
Can this check be removed again since you have added `fsck.bcachefs`?

Reply:
No, it's just a symlink to the binary. fsck is not implemented yet.